### PR TITLE
chore: update workflow & job names

### PR DESCRIPTION
**Pull request by nr-kkenney**
* updated workflow & job names to better describe what is happening, and better indicate if a particular check can be ignored.

Resolves: #2613

### CODE REVIEW

File: .github/workflows/pr-triage.yml [hunk 1]
```diff
@@ -10,11 +10,11 @@ env:
 
 jobs:
   content_project:
+    name: did the pr get added to the docs github board? (optional)
```
Comment (jpvajda) [inline .github/workflows/pr-triage.yml:13]: why is this framed as question? 

```suggestion
    name: pr added to the docs github board (optional)
```

File: .github/workflows/pr-triage.yml [hunk 2]
```diff
@@ -25,11 +25,11 @@ jobs:
         gh api -H "$HEADER" -X POST projects/columns/$COLUMN/cards -f content_type='PullRequest' -F content_id=$PR_ID
 
   localization_project:
+    name: did the pr get added to the localization github board? (optional)
```
Comment (jpvajda) [inline .github/workflows/pr-triage.yml:28]: same feedback

```suggestion
    name: pr added to the localization github board (optional)
```

File: .github/workflows/update-whats-new-ids.yml [hunk 3]
```diff
@@ -11,6 +11,7 @@ env:
 
 jobs:
   update-whats-new-ids:
+    name: prepare in-product anncouncements (optional)
```
Comment (jpvajda) [inline .github/workflows/update-whats-new-ids.yml:14]:
```suggestion
    name: generates in-product anncouncements (optional)
```

File: .github/workflows/validate-pr.yml [hunk 4]
```diff
@@ -12,7 +12,7 @@ env:
 
 jobs:
   build:
-    name: Gatsby Build
+    name: did the docs sites build?
```
Comment (jpvajda) [inline .github/workflows/validate-pr.yml:15]:
```suggestion
    docs sites build process
```

File: .github/workflows/validate-pr.yml [hunk 5]
```diff
@@ -51,7 +51,7 @@ jobs:
           CI: true
 
   test:
-    name: Run Tests
+    name: did tests succeed?
```
Comment (jpvajda) [inline .github/workflows/validate-pr.yml:54]:
```suggestion
    name: all tests succeeded
```

File: .github/workflows/validate-pr.yml [hunk 6]
```diff
@@ -77,7 +77,7 @@ jobs:
         run: yarn test --passWithNoTests
 
   lint:
-    name: Run Eslint
+    name: did the linter succeed?
```
Comment (jpvajda) [inline .github/workflows/validate-pr.yml:80]:
```suggestion
    name: all linter checks succeeded
```

**Review by jpvajda — CHANGES_REQUESTED** [review]

**Comment by nr-kkenney** [comment]
@jpvajda feedback came from @paperclypse to frame most of the checks as questions. @paperclypse if you are good with john's suggestions, I can make those changes.

**Comment by jpvajda** [comment]
@nr-kkenney Thanks for clarifying I'd prefer we follow a declarative approach and just name the jobs as statements as using questions in programming isn't common. cc @paperclypse

**Comment by jpvajda** [comment]
@nr-kkenney I haven't heard back from @paperclypse but I want to move this forward, so let's go with my suggested edits as this is a more declarative approach.

**Review by jpvajda — APPROVED** [review]
LGTM

**Comment by paperclypse** [comment]
@jpvajda Just because something isn't common, doesn't mean it shouldn't be done :) But I think it's fine in this case. Anything is an improvement over what we had, basically. Thanks!

**Pull request merged**